lib/topological_inventory/providers/common/operations/source.rb

```diff
@@ -106,10 +114,15 @@ def connection_check
         def update_source_and_subresources(status, error_message = nil)
           logger.availability_check("Updating source [#{source_id}] status [#{status}] message [#{error_message}]")
 
-          update_source(status)
-
-          update_endpoint(status, error_message) if endpoint
-          update_application(status) if application
+          if checks_via_kafka
```

Comment:
~Tying in with my other comment, I think we should just update the `update_source`, `update_endpoint`, and `update_application` methods to use kafka.~ Just need to do it eventually - but it'll be a refactor.